hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/WALEntryFilterRetryableException.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.replication.regionserver;
+
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * This exception should be thrown from any wal filter when the filter is expected
+ * to recover from the failures and it wants the replication to backup till it fails.
+ * There is special handling in replication wal reader to catch this exception and
+ * retry.
+ */
+@InterfaceAudience.Public
```

Comment:
InterfaceAudience should be LimitedPrivate(REPLICATION)

Reply:
@gjacoby126 I believe this will be used by wal filters outside of hbase codebase. 
Will `LimitedPrivate` allow it?
BTW, ReplicationException has public too **https://github.com/apache/hbase/blob/master/hbase-client/src/main/java/org/apache/hadoop/hbase/replication/ReplicationException.java#L29**

Reply:
LimitedPrivate only please. This signals how we intend to support it. Public means it is for general use and we will keep it very stable. LP(REPLICATION) seems more than appropriate because this is an internal implementation detail that has become part of the interface contract via this exception.